src/incubator/Slider/index.tsx

```diff
@@ -0,0 +1,575 @@
+        
+        const outputRange = [0, trackSize.value.width];
+        const inputRange = 
+          minimumValue < 0 ? [Math.abs(maximumValue), Math.abs(minimumValue)] : [minimumValue, maximumValue];
+        const stepInterpolated = 
+          interpolate(stepXValue.value, inputRange, outputRange);
+        
+        const newX = Math.round(x / stepInterpolated) * stepInterpolated; // getStepComputedX(x) - worklet error
+        runOnJS(updateDefaultThumb)(newX);
+      }
+    });
+
+  const rangeThumbGesture = Gesture.Pan()
+    .onBegin(() => {
+      if (disabled) {
+        return;
+      }
+
+      isPressedRange.value = true;
+      activeThumb.value = ThumbType.RANGE;
+    })
+    .onUpdate((e) => {
+      if (disabled) {
+        return;
+      }
+
+      onSeekStart?.();
+
+      let newX = rangeThumbStart.value.x + e.translationX * (shouldDisableRTL ? 1 : rtlFix);
+      if (newX > trackSize.value.width) { // top edge
+        newX = trackSize.value.width;
+      }
+      if (newX >= defaultThumbStart.value.x + rangeGap && newX <= trackSize.value.width) { // range
+        rangeThumbOffset.value = {
+          x: newX,
+          y: 0
+        };
+
+        activeTrackWidth.value = rangeThumbOffset.value.x - defaultThumbStart.value.x;
+        
+        runOnJS(onChange)(useRange ? {min: defaultThumbStart.value.x, max: newX} : newX);
+      }
+    })
+    .onEnd(() => {
+      onSeekEnd?.();
+
+      rangeThumbStart.value = {
+        x: rangeThumbOffset.value.x,
+        y: 0
+      };
+    })
+    .onFinalize(() => {
+      isPressedRange.value = false;
+
+      if (shouldBounceToStep) {
+        const x = rangeThumbOffset.value.x;
+        
+        const outputRange = [0, trackSize.value.width];
+        const inputRange = 
+          minimumValue < 0 ? [Math.abs(maximumValue), Math.abs(minimumValue)] : [minimumValue, maximumValue];
+        const stepInterpolated = 
+          interpolate(stepXValue.value, inputRange, outputRange);
+
+        const newX = Math.round(x / stepInterpolated) * stepInterpolated; // getStepComputedX(x) - worklet error
+        
+        runOnJS(updateRangeThumb)(newX);
+      }
+    });
+
+  const getStepComputedX = (x: number) => {
+    'worklet';
+    const outputRange = [0, trackSize.value.width];
+    const inputRange = 
+      minimumValue < 0 ? [Math.abs(maximumValue), Math.abs(minimumValue)] : [minimumValue, maximumValue];
+    const stepInterpolated = Math.abs(interpolate(stepXValue.value, inputRange, outputRange));
+    return Math.round(x / stepInterpolated) * stepInterpolated;
+  };
+
+  const trackAnimatedStyles = useAnimatedStyle(() => {
+    if (useRange) {
+      return {
+        transform: [
+          {translateX: defaultThumbOffset.value.x * rtlFix}
+        ],
+        width: activeTrackWidth.value
+      };
+    } else {
+      return {
+        width: activeTrackWidth.value
+      };
+    }
+  });
+
+  /** renders */
+
+  const renderRangeThumb = () => {
```

Comment:
The default thumb and range thumb are pretty similar and can be extracted to a separate sub component

Reply:
I know. The animated styles are not updated properly when I pass them as a parameter...